pkg/tsdb/sql_engine.go

```diff
@@ -163,6 +172,26 @@ func (e *sqlQueryEndpoint) Query(ctx context.Context, dsInfo *models.DataSource,
 	return result, nil
 }
 
+// global macros/substitutions for all sql datasources
+func Interpolate(query *Query, timeRange *TimeRange, sql string) (string, error) {
+	rExp, _ := regexp.Compile(`\$__(interval|interval_ms)\b`)
+
+	sql = ReplaceAllStringSubmatchFunc(rExp, sql, func(groups []string) string {
+		switch groups[1] {
+		case "interval":
```

Comment:
There's existing functions and interfaces for calculating intervals that you should use, see influxdb code [here](https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/model_parser.go#L42) and [here](https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/query.go#L37-L38). There should be similar example code for prometheus and elasticsearch.

Sorry, should have spotted this earlier.